### ISSUE TITLE
Gateway-MT: Re-evaluate logging

### ISSUE THREAD

**Issue by wthorp** [storj/edge#31]
We currently use a global Sync.Once mutex with the Minio logger.  The need to copy things between context and use of globals is unfortunate.  Could we simplify the logging solution if we use our own middleware?  Maybe we would have to implement a custom RegisterAPIRouter equivalent; Bill had one at one point that included all the routes and was hand written, not derived from Minio.

**Comment by ferristocrat** [comment]
@amwolff @halkyon - is this still relevant?

**Comment by halkyon** [comment]
@ferristocrat Yes, there is still a need to clean this up by removing the minio dependency in our logging/metrics middleware.

**Comment by amwolff** [comment]
we need to get rid of minio, so closing this anyway
